sdk/core/switcher/entry.S

```diff
@@ -415,6 +425,13 @@ exception_entry_asm:
 	li          a1, 0
 	// If we don't have enough space, give up and force unwind
 	bltu        t1, tp, .Lforce_unwind
+	// make sure the caller's CSP is valid
+	cgettag     t1, ct0
+	beqz        t1, .Lforce_unwind
+	// make sure the caller's CSP has the expected permissions
+	cgetperm    t1, ct0
+	li 	        tp, 0x7e
+	bne	        tp, t1, .Lforce_unwind
```

Comment:
These checks are the same as above, can you make them a macro? We should also add a sealing check.

Reply:
Great point, thanks!